### PR TITLE
🚧 Add the calendarItemTapped action

### LINKED ISSUE #204: Add the ItemTapped action
## Description

Add the ItemTapped action and handle it in the reducer. This should simply set the selected item in the state.

## Definition of done
- [ ] The `selectedItem` is set in the state with an editable time entry
- [ ] Tests are created

### PR DESCRIPTION
**Pull request by juanlaube**
## Info
Adds the action to select calendar items.

### Summary
Adds the calendarItemTapped action and sets the selectedItem in the reducer.

### Technical considerations
Nothing remarkable to add here.

### Relationships
Closes #204 

### Pull Request checklist
- [x] The code does not hinder the app's performance
- [x] The code does not introduce regressions
- [x] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests
- [ ] The code used custom templates

### Review template

- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones

- [ ] The code considers light and dark theme
- [ ] The code considers RTL and does not mention left/right
- [ ] The code considers i18n
  - [ ] There are no hardcoded strings in the UI
  - [ ] All strings are included in every language we support OR there's an issue requesting new translations
- [ ] There are no hard-coded numbers referring to dimensions or font sizes
- [ ] The code considers UI elements that are similar and extracts their similarities into styles to avoid duplication

## Merge permissions
🦑

### CODE REVIEW

File: Modules/Calendar/CalendarTests/CalendarDayTests/CalendarDayReducerTests.swift [hunk 1]
```diff
@@ -205,4 +206,75 @@ class CalendarDayReducerTests: XCTestCase {
             Step(.send, CalendarDayAction.emptyPositionLongPressed(now))
         )
     }
+
+    func test_itemTapped_withATimeEntry() {
```
Comment (dshr) [inline Modules/Calendar/CalendarTests/CalendarDayTests/CalendarDayReducerTests.swift:210]:
```suggestion
    func test_itemTapped_withATimeEntry_selectsTimeEntry() {
```

File: Modules/Calendar/CalendarTests/CalendarDayTests/CalendarDayReducerTests.swift [hunk 2]
```diff
@@ -205,4 +206,75 @@ class CalendarDayReducerTests: XCTestCase {
...
+            initialState: state,
+            reducer: reducer,
+            steps:
+            Step(.send, .itemTapped(calendarItem)) {
+                $0.selectedItem = .left(EditableTimeEntry.fromSingle(timeEntry))
+            }
+        )
+    }
+
+    func test_itemTapped_withACalendarEvent() {
```
Comment (dshr) [inline Modules/Calendar/CalendarTests/CalendarDayTests/CalendarDayReducerTests.swift:233]:
```suggestion
    func test_itemTapped_withACalendarEvent_selectsCalendarEvent() {
```

File: Modules/Calendar/CalendarTests/CalendarDayTests/CalendarDayReducerTests.swift [hunk 3]
```diff
@@ -205,4 +206,75 @@ class CalendarDayReducerTests: XCTestCase {
...
+            initialState: state,
+            reducer: reducer,
+            steps:
+            Step(.send, .itemTapped(calendarItem)) {
+                $0.selectedItem = .right(calendarEvent)
+            }
+        )
+    }
+
+    func test_itemTapped_selectsAnotherItem() {
```
Comment (dshr) [inline Modules/Calendar/CalendarTests/CalendarDayTests/CalendarDayReducerTests.swift:256]:
```suggestion
    func test_itemTapped_withAlreadySelectedItem_selectsTheNewItem() {
```

File: Modules/Calendar/Calendar/CalendarDay/CalendarDayAction.swift [hunk 4]
```diff
@@ -24,6 +25,9 @@ extension CalendarDayAction: CustomDebugStringConvertible {
 
         case .emptyPositionLongPressed(let date):
             return "EmptyPositionLongPressed: \(date)"
+
+        case .itemTapped:
+            return "ItemTapped"
```
Comment (sofokli1) [inline Modules/Calendar/Calendar/CalendarDay/CalendarDayAction.swift:30]: no item logged?

Comment (juanlaube) [reply Modules/Calendar/Calendar/CalendarDay/CalendarDayAction.swift:30]: I'm not sure what we could log that is relevant 🤔

Comment (sofokli1) [reply Modules/Calendar/Calendar/CalendarDay/CalendarDayAction.swift:30]: start, duration, and description seem like good info to have in debug but I leave it up to you

**Pull request merged**